cypress-io/cypress
# chore: update electron notary tool to v2

- Closes N/A

### Additional details
Updates the `electron-notarize` tool from `v1` to `v2` which goes from the `legacy` notary tool (which will be unsupported and will not work starting 11/23) to the new `notarytool`. I upgraded this on a branch, which ran on the macstadium box and it looks to not be able to use `v1` any longer (not sure why). I figured the quickest solution is to just bump the notary tool 😅 .

FWIW the new notary tool is supposed to be 10x faster as described in the [repo](https://github.com/electron/notarize#method-notarizeopts-promisevoid). To use the new `notarytool`, the `teamId` property is needed, which has been correctly added to the circle context.

### Steps to test
Check to make sure `darwin_x64` and `darwin_arm64` jobs in circle are green and passing and notarization works when building the binary

### How has the user experience changed?


### PR Tasks
